frame/transaction-payment/src/lib.rs

```diff
@@ -331,41 +333,40 @@ impl<T: Config> Module<T> where
 		RuntimeDispatchInfo { weight, class, partial_fee }
 	}
 
+	/// Query the detailed fee of a given `call`.
+	pub fn query_fee_details<Extrinsic: GetDispatchInfo>(
+		unchecked_extrinsic: Extrinsic,
+		len: u32,
+	) -> FeeDetails<BalanceOf<T>>
+	where
+		T: Send + Sync,
+		BalanceOf<T>: Send + Sync,
+		T::Call: Dispatchable<Info=DispatchInfo>,
+	{
+		let dispatch_info = <Extrinsic as GetDispatchInfo>::get_dispatch_info(&unchecked_extrinsic);
+		Self::compute_fee_details(len, &dispatch_info, 0u32.into())
+	}
+
 	/// Compute the final fee value for a particular transaction.
-	///
-	/// The final fee is composed of:
```

Comment:
I am fine with this documentation being removed, but I would add it _somewhere_ as it is useful. Maybe module doc?

Reply:
Actually I spread the docs to the various places, but I think it's ok to also add them to the module doc :(, please take a look at the latest commit.